python/lsst/faro/measurement/TractMeasurementTasks.py

```diff
@@ -51,15 +51,14 @@ def run(
         self, metricName, catalogs, photoCalibs=None, astromCalibs=None, dataIds=None
     ):
         self.log.info("Measuring %s", metricName)
-        bands = set([f["band"] for f in dataIds])
 
-        if ("g" in bands) & ("r" in bands) & ("i" in bands):
-            rgicatAll = makeMatchedPhotom(dataIds, catalogs, photoCalibs)
+        if ("g" in catalogs) & ("r" in catalogs) & ("i" in catalogs):
```

Comment:
I don't yet understand how the `catalogs` contains a string that indicates the band.

Reply:
The `&` here seems suspicious as well. Shouldn't that be `and`?

Also, for the previous version since `bands` was a `set` that could have been written much more concisely by:

```python
required_bands = set("gri")
if required_bands.issubset(bands):
    ...
```

Reply:
It doesn't actually work; I was hasty in testing this and have re-worked it significantly.